docs/functionaldatatypes/these.md

```diff
@@ -0,0 +1,57 @@
+---
+id: these
+title: "These"
+---
+
+`These[A, B]` is a data type that models a value that can be either a `Left` with a value of type `A`, a `Right` with a value of type `B`, or a `Both` with both a value of type `A` and a value of type `B`.
+
+```scala mdoc
+sealed trait These[+A, +B] {
+  case class Left[+A](a: A) extends These[A, Nothing]
+  case class Right[+B](b: B) extends These[Nothing, B]
+  case class Both[+A, +B](a: A, b: B) extends These[A, B]
+}
+```
+
+You can think of `These` as like `Either` except that it has one additional case. Whereas an `Either` contains either an `A` or a `B` but never both, a `These` can contain both an `A` and a `B`.
+
+A simple example of a situation where `These` might arise is in merging two upstreams in a streaming application. We could be signaled when at least one upstream has data for us and could have a situation where either only the left upstream has data for us, only the right upstream has data for us, or both upstreams have data for us.
+
+Another example of using the `These` data type is in implementing the `zipAll` operator on collections.
```

Comment:
you could clarify with wording use "hypothetical implementation" since ZIO `Chunk` actually uses Builder and not  `These`

Reply:
Clarified to "implement a variant of".